tests/migrations/test_commands.py

```diff
@@ -1427,6 +1427,48 @@ def test_prune_no_app_label(self):
         with self.assertRaisesMessage(CommandError, msg):
             call_command("migrate", prune=True)
 
+    @override_settings(
+        MIGRATION_MODULES={
+            "migrations": "migrations.test_migrations_squashed_no_replaces",
+            "migrations2": "migrations2.test_migrations_2_squashed",
+        },
+    )
+    def test_prune_specified_app_other_irrelevant_app_with_replaces(self):
+        """
+        Prune only the specified app migrations
+        """
+        recorder = MigrationRecorder(connection)
+        recorder.record_applied("migrations", "0001_initial")
+        recorder.record_applied("migrations", "0002_second")
+        recorder.record_applied("migrations", "0001_squashed_0002")
+        # Second app has squashed migrations with replaces
+        # which should not be checked and noe error should occur.
+        recorder.record_applied("migrations2", "0001_initial")
+        recorder.record_applied("migrations2", "0002_second")
+        recorder.record_applied("migrations2", "0001_squashed_0002")
+        out = io.StringIO()
+        try:
+            call_command("migrate", "migrations", prune=True, stdout=out, no_color=True)
+            self.assertEqual(
+                out.getvalue(),
+                "Pruning migrations:\n"
+                "  Pruning migrations.0001_initial OK\n"
+                "  Pruning migrations.0002_second OK\n",
+            )
+            applied_migrations = [
+                migration
+                for migration in recorder.applied_migrations()
+                if migration[0] == "migrations"
+            ]
+            self.assertEqual(applied_migrations, [("migrations", "0001_squashed_0002")])
+        finally:
+            recorder.record_unapplied("migrations", "0001_initial")
+            recorder.record_unapplied("migrations", "0001_second")
+            recorder.record_unapplied("migrations", "0001_squashed_0002")
+            recorder.record_applied("migrations2", "0001_initial")
+            recorder.record_applied("migrations2", "0002_second")
+            recorder.record_applied("migrations2", "0001_squashed_0002")
+
```

Comment:
Do we need a new test app? I think we could the same as in `test_prune_deleted_squashed_migrations_in_replaces`, e.g.
````
@override_settings(
    MIGRATION_MODULES={
        'migrations': 'migrations.test_migrations_squashed',
        'migrations2': 'test_migrations_squashed_no_replaces',
    })
    ....
```

Reply:
We are going to test the scenario that we have 2 apps and one of them has squashed migration with replaces (migrations2) but we only want to prune the first app migrations.